app/api/dataset-download/route.ts

```diff
@@ -0,0 +1,155 @@
+import { authOptions } from "@/lib/auth/options";
+import prisma from "@/lib/prisma";
+import { getServerSession } from "next-auth";
+import { redirect } from "next/navigation";
+import { NextRequest, NextResponse } from "next/server";
+import json2csv from 'json2csv';
+
+export async function GET(req: NextRequest) {
+  try {
+    const session = await getServerSession(authOptions);
+    if (!session || !session.user) {
+      redirect("/login");
+    }
+
+    const id = req.nextUrl.searchParams.get("id") as string;
+    const datasetId = req.nextUrl.searchParams.get("dataset_id") as string;
+    const pageParam = req.nextUrl.searchParams.get("page");
+    let page = pageParam ? parseInt(pageParam, 10) : 1;
+    const pageSizeParam = req.nextUrl.searchParams.get("pageSize");
+    const pageSize = pageSizeParam ? parseInt(pageSizeParam, 10) : 500;
+
+    if (!datasetId && !id) {
+      return NextResponse.json(
+        {
+          message: "No dataset id or project id provided",
+        },
+        { status: 404 }
+      );
+    }
+
+    if (datasetId) {
```

Comment:
this if is unnecessary since you already check and returned on line 22

Reply:
this is checking if the 'prisma.dataset.findFirst({ ... })' query retrives a 'dataset'. If it is empty, then we return a NextResponse saying 'no dataset found'